src/client/input_event_adapter.cpp

```diff
@@ -0,0 +1,467 @@
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::MoveLeft)), "Fly left", hashToString(hashGlfwKeyInput(GLFW_KEY_A, 0))});
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::MoveBackward)), "Fly backward", hashToString(hashGlfwKeyInput(GLFW_KEY_S, 0))});
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::MoveDown)), "Fly down", hashToString(hashGlfwKeyInput(GLFW_KEY_C, 0))});
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::MoveUp)), "Fly up", hashToString(hashGlfwKeyInput(GLFW_KEY_SPACE, 0))});
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::RollLeft)), "Roll counterclockwise", hashToString(hashGlfwKeyInput(GLFW_KEY_Q, 0))});
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::RollRight)), "Rool clockwise", hashToString(hashGlfwKeyInput(GLFW_KEY_E, 0))});
+	s.push_back({
+		player_section, std::string(actionToString(PlayerActionEvents::PauseGame)), "Pause movement and release cursor",
+			hashToString(hashGlfwMouseKeyInput(GLFW_MOUSE_BUTTON_LEFT, 0)) + std::string(KEYS_SEQUENCE_DELIMITER) + hashToString(hashGlfwKeyInput(GLFW_KEY_ESCAPE, 0))
+	});
+
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::IncreaseSpeed)), "Incrase movement speed", hashToString(hashGlfwMouseScrollInput(1.0))});
+	s.push_back({player_section, std::string(actionToString(PlayerActionEvents::DecreaseSpeed)), "Decrease movement speed", hashToString(hashGlfwMouseScrollInput(-1.0))});
+
+	return s;
+}
+
+uint32_t InputEventAdapter::hashGlfwKeyInput(int key, int mods) {
+	uint32_t hash = 0U;
+
+	// Get 9 bits, because this is actually a range of GLFW `key`
+	// More info: https://www.glfw.org/docs/3.3.2/group__keys.html
+	hash = key & 0x1FF;
+
+	// `mods` is flags and have wide in 2*4 = 8 bit
+	// More info: https://www.glfw.org/docs/3.3.2/group__mods.html
+	hash |= (mods & 0xFF ) << 9;
+
+	// And we have three type of input: from keyboard, mouse and scroll.
+	// So, use first two bytes for GLFW event type
+	// 00 - keyboard, 01 - mouse, 10 - scroll, 11 - invalid hash
+
+	// Summary: 9 + 8 + 2 = 19 bits
+	return hash;
+}
+
+uint32_t InputEventAdapter::hashGlfwMouseKeyInput(int button, int mods) {
+	// See hashGlfwKeyInput and GLFW event type explanation
+	uint32_t hash = 0x40000000; // This is 30^2
+
+	// `button` have only 3 significant bits, becuase GLFW have only 8 mouse group__keys
+	// More info: https://www.glfw.org/docs/3.3.2/group__buttons.html
+	hash |= button & 0x7;
+
+	// See explanation for `mods` in hashGlfwKeyInput
+	hash |= (mods & 0xFF) << 3;
+
+	// Summary: 3 + 8 + 2 = 13 bits
+	return hash;
+}
+
+uint32_t InputEventAdapter::hashGlfwMouseScrollInput(double yoffset) {
+	// See hashGlfwKeyInput and GLFW event type explanation
+	uint32_t hash = 0x80000000; // This is 31^2
```

Comment:
31^2? Did you mean 2^31? This value can also be explicitly expressed as `1 << 31`.